add si32 HAL

M1cha opened the pull request:
Source: precision32-development-environment-offline.exe
	sha512: 6bccef30995da803985fcdfa4065c45e00e0eeefc300d61b7215926a97997dc8ce45078ffeb349602340759650f30aeccf3b1aa50086c61edc2426d016063cae
	Website: https://www.silabs.com/developers/precision32-32-bit-microcontroller-development-suite?tab=downloads

rettichschnidi commented on si32/README:
```diff
@@ -0,0 +1,50 @@
++++++++++++++
+
+  * Download precision32-development-environment-offline.exe
+  * run `./scripts/extract_hal /path/to/downloaded-file`
```
It does not seem to work for me:

```shell
$ sha512sum ~/Downloads/precision32-development-environment-offline.exe
6bccef30995da803985fcdfa4065c45e00e0eeefc300d61b7215926a97997dc8ce45078ffeb349602340759650f30aeccf3b1aa50086c61edc2426d016063cae  /home/reto/Downloads/precision32-development-environment-offline.exe
$ scripts/extract_hal ~/Downloads/precision32-development-environment-offline.exe

7-Zip (a) [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=en_US.utf8,Utf16=on,HugeFiles=on,64 bits,32 CPUs AMD Ryzen 9 5900X 12-Core Processor             (A20F10),ASM,AES-NI)

Scanning the drive for archives:
1 file, 427228821 bytes (408 MiB)

Extracting archive: /home/username/Downloads/precision32-development-environment-offline.exe

ERRORS:
Headers Error
Unconfirmed start of archive


WARNINGS:
Headers Error
There are data after the end of archive

--
Path = /home/username/Downloads/precision32-development-environment-offline.exe
Type = zip
ERRORS:
Headers Error
Unconfirmed start of archive
WARNINGS:
Headers Error
There are data after the end of archive
Offset = 1815905
Physical Size = 271798
Tail Size = 425141118


No files to process


Archives with Errors: 1

Warnings: 1

Open Errors: 1
```

Any ideas?

M1cha replied:
You're using 7zip 16.02 which was released in 2016. I'm using [17.04](https://github.com/p7zip-project/p7zip/releases/tag/v17.04) which has support for lots of new file formats. (There's also 17.05 which got released recently which I haven't tested, yet).

M1cha replied:
I documented the version requirement.

rettichschnidi commented on si32/README:
```diff
@@ -0,0 +1,50 @@
+Silicon Labs SI32 HAL
```
```suggestion
Silicon Labs Si32 HAL
```

M1cha replied:
done

M1cha commented on si32/README:
```diff
@@ -0,0 +1,50 @@
+
+   The SDK contains an incompatible, proprietary license in `Documentation/Silabs_License_Agreement.txt`,
+   but all of the source code links to a newer, permissive license [0].
+   We have official, public confirmation [1], that the newer license applies.
```
good idea

fkokosinski commented on si32/si32Hal/CPU/core_cm3.h:
```diff
@@ -0,0 +1,1236 @@
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE APPLY TO THIS SOFTWARE.
+ * ARM SHALL NOT, IN ANY CIRCUMSTANCES, BE LIABLE FOR SPECIAL, INCIDENTAL, OR
+ * CONSEQUENTIAL DAMAGES, FOR ANY REASON WHATSOEVER.
+ *
```
Wrong license? Keep in mind that Zephyr accepts only OSI-compliant licenses for HAL source files.

M1cha replied:
These are actually not needed because zephyr provides them already and we can just use those instead.
But what makes a license OSI-compliant? Is the permissive `Silicon Laboratories End User License Agreement` still an issue?

pgielda replied:
See here for a list: https://opensource.org/licenses/

M1cha replied:
Okay that's unfortunate since it means that [this okay](https://github.com/zephyrproject-rtos/zephyr/issues/55287#issuecomment-1546199849) is not enough.
@silabs-joe Is it possible for you to allow us to change the license to z-lib(the same license as the gecko hal)?
We could change it ourselves so you don't have to put in any work, but we do obviously need the okay.

silabs-spurthi replied:
Hi @M1cha,
I am a Silicon Labs employee. I've checked with our legal team and they approve Gardena applying a ZLIB license to the code in the si32Hal folder for the Zephyr OS project. So you can adapt or modify the files to include the ZLIB license instead of the current license text. 

Below is the ZLIB license text to use in the headers for the si32Hal folder:
**_Copyright 2012 (c) Silicon Laboratories Inc._
 
_This siHAL software is provided 'as-is', without any express or implied warranty.  In no event will the authors be held liable for any damages arising from the use of this software._
 
_Permission is granted to anyone to use this software for any purpose, including commercial applications, and to alter it and redistribute it freely, subject to the following restrictions:_
 
_1. The origin of this software must not be misrepresented; you must not claim that you wrote the original software. If you use this software in a product, an acknowledgment in the product documentation would be appreciated but is not required._
 
_2. Altered source versions must be plainly marked as such, and must not be misrepresented as being the original software.
 
3. This notice may not be removed or altered from any source distribution._**

Please let me know if you have any questions.

Best regards,
Spurthi

mmahadevan108 replied:
It would good to include the `SPDX license identifier` as well so its easy to see the license type.

M1cha commented on si32/README:
```diff
@@ -0,0 +1,50 @@
+
+  * Download precision32-development-environment-offline.exe
+  * run `./scripts/extract_hal /path/to/downloaded-file`
+  * The extracted HAL will be in `$CWD/tmp`, Copy the necessary directories
```
nope. after a `,` you don't start with an uppercase character. Also, that sentence is gone now.

M1cha commented on si32/README:
```diff
@@ -0,0 +1,50 @@
+   v1.1.2
+
+Purpose:
+   Add support for Silican Labs sim3u1xx SoCs
```
done

M1cha commented on si32/README:
```diff
@@ -0,0 +1,50 @@
+   Add support for Silican Labs sim3u1xx SoCs
+
+Description:
+   This code component is used to add Zephyr support for Silicon Labs sim3u1xx SoCs.
```
done

M1cha commented on si32/README:
```diff
@@ -0,0 +1,50 @@
+   v1.1.2
+
+Purpose:
+   Add support for Silican Labs sim3u1xx SoCs
```
done

M1cha commented:
I've done the following:
- rebased to `master`
- addressed all comments
- added a copy_hal script which copies and patches the SDK
- added a dump of this site in license-change-permit.html so the relicensing permit will not be lost.
- I excluded the CMSIS headers since they had a proprietary license and we have a compatible version in `modules/hal/cmsis`.
- I excluded startup and linker files since they had a proprietary license and we don't need them.

I did not add a SPDX license identifier since @silabs-spurthi never gave their ok for that.

silabs-Joe commented:
@M1cha Our legal team has approved adding the SPDX license identifier.

rettichschnidi reviewed (approved):
LGTM. Will look even better with SPDX identifiers.

M1cha commented:
@msierszulski @fkokosinski I've added the SPDX-identifiers, updated the copy of this page and extended the commit description.

fkokosinski reviewed (approved):
Hi @M1cha! Thanks, looks good now. Could you create a PR that uses these changes, as the one that you created before is now closed?

We usually require the Zephyr PR to be ACK'd before merging PRs in HAL repos. I'll merge this PR when that happens.

rettichschnidi commented:
Now that #66 got merged, this PR can be closed.
